File: geomesa-utils-parent/geomesa-utils/src/main/java/org/locationtech/geomesa/utils/interop/WKTUtils.java
/***********************************************************************
 * Copyright (c) 2013-2023 Commonwealth Computer Research, Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Apache License, Version 2.0
 * which accompanies this distribution and is available at
 * http://www.opensource.org/licenses/apache2.0.php.
 ***********************************************************************/

package org.locationtech.geomesa.utils.interop;

import org.locationtech.geomesa.utils.text.WKTUtils$;
import org.locationtech.jts.geom.Geometry;

public class WKTUtils {
    public static Geometry read(String WKTgeometry) {
        return WKTUtils$.MODULE$.read(WKTgeometry);
    }

    public static String write(Geometry g) {
        return WKTUtils$.MODULE$.write(g);
    }
}

File: geomesa-utils-parent/geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/text/WKTUtils.scala
/***********************************************************************
 * Copyright (c) 2013-2025 General Atomics Integrated Intelligence, Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Apache License, Version 2.0
 * which accompanies this distribution and is available at
 * https://www.apache.org/licenses/LICENSE-2.0
 ***********************************************************************/

package org.locationtech.geomesa.utils.text

import org.locationtech.jts.geom.Geometry
import org.locationtech.jts.io.{WKTReader, WKTWriter}

object WKTUtils extends WKTUtils

trait WKTUtils {
  private[this] val readerPool = new ThreadLocal[WKTReader]{
    override def initialValue = new WKTReader
  }
  private[this] val writerPool = new ThreadLocal[WKTWriter]{
    override def initialValue = new WKTWriter
  }

  def read(s: String): Geometry = readerPool.get.read(s)
  def write(g: Geometry): String = writerPool.get.write(g)
}
